src/ResourceManager/Sql/Commands.Sql/ServerDnsAlias/Cmdlet/RemoveAzureSqlServerDnsAlias.cs

```diff
@@ -0,0 +1,158 @@
+﻿// ----------------------------------------------------------------------------------
+//
+// Copyright Microsoft Corporation
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+// http://www.apache.org/licenses/LICENSE-2.0
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+// ----------------------------------------------------------------------------------
+
+using System.Collections.Generic;
+using System.Globalization;
+using System.Management.Automation;
+using Microsoft.Azure.Management.Internal.Resources.Utilities.Models;
+
+namespace Microsoft.Azure.Commands.Sql.ServerDnsAlias.Cmdlet
+{
+	[Cmdlet(VerbsCommon.Remove, "AzureRmSqlServerDnsAlias", SupportsShouldProcess = true, ConfirmImpact = ConfirmImpact.Medium)]
+	[OutputType(typeof(Model.AzureSqlServerDnsAliasModel))]
+	public class RemoveAzureSqlServerDNSAlias : AzureSqlServerDnsAliasCmdletBase
+	{
+		protected const string CmdletParametersParameterSet =
+			"Remove a Server Dns Alias from cmdlet input parameters";
+
+		protected const string ObjectParameterSet =
+			"Remove a Server Dns Alias from AzureSqlServerDnsAliasModel instance definition";
+
+		protected const string ResourceIdParameterSet =
+			"Remove a Server Dns Alias from an Azure resource id";
+
+		/// <summary>
+		/// Gets or sets the name of the server dns alias to remove
+		/// </summary>
+		[Parameter(ParameterSetName = CmdletParametersParameterSet,
+			Mandatory = true,
+			HelpMessage = "Azure Sql Server Dns Alias name")]
+		[Alias("Name")]
+		[ValidateNotNullOrEmpty]
+		public string DnsAliasName { get; set; }
+
+		/// <summary>
+		/// Gets or sets the name of the Azure Sql Server to use
+		/// </summary>
+		[Parameter(ParameterSetName = CmdletParametersParameterSet,
+			Mandatory = true,
+			HelpMessage = "The Azure Sql Server name.")]
+		[ValidateNotNullOrEmpty]
+		public override string ServerName { get; set; }
+
+		/// <summary>
+		/// Gets or sets the name of the resource group to use.
+		/// </summary>
+		[Parameter(ParameterSetName = CmdletParametersParameterSet,
+			Mandatory = true,
+			ValueFromPipelineByPropertyName = true,
```

Comment:
@maboja-msft please remove this line - `ResourceId` is the only parameter that should accept its value from the pipeline by property name